pkg/container-collection/options.go

```diff
@@ -788,6 +799,63 @@ func WithOCIConfigEnrichment() ContainerCollectionOption {
 	}
 }
 
+// WithOCIConfigForInitialContainer enriches initial containers with the OCI config.
+// It assumes that the initial containers are already populated in the ContainerCollection
+func WithOCIConfigForInitialContainer() ContainerCollectionOption {
+	return func(cc *ContainerCollection) error {
+		for _, container := range cc.initialContainers {
+			ppid := 0
+			sPath := filepath.Join(host.HostProcFs, fmt.Sprintf("%d", container.ContainerPid()), "status")
+			sFile, err := os.Open(sPath)
+			if err != nil {
+				log.Errorf("OCIConfig enricher: failed to open status file for container %s: %s", container.Runtime.ContainerID, err)
+				continue
+			}
+			defer sFile.Close()
+
+			// Read the status file to get the parent pid
+			reader := bufio.NewReader(sFile)
+			for {
+				line, err := reader.ReadString('\n')
+				if err != nil {
+					log.Errorf("OCIConfig enricher: failed to read status file for container %s: %s", container.Runtime.ContainerID, err)
+					break
+				}
```

Comment:
Need to handle `io.EOF`.
reader.ReadString() can return `io.EOF` together with a non-empty `line`.

I wonder if ioutil.ReadFile is easier.

Reply:
I adopted the implementation, Let me know if it looks good now!